app/services/place_hold_form/builder.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# Responsible for building data needed to populate the place hold form.
+class PlaceHoldForm::Builder
+  def initialize(catkey:, user_token:, client:)
+    @catkey = catkey
+    @user_token = user_token
+    @client = client # SymphonyClient
+    @volumetric_calls = []
+  end
+
+  def generate
+    bib_info = Bib.new(parsed_symphony_response(:get_bib_info, @catkey, @user_token))
+    @call_list = bib_info.call_list.map { |c| Call.new record: c }
+    @holdable_locations = find_holdable_locations
```

Comment:
```suggestion
    find_holdable_locations
```

What about setting `@holdable_locations` with find_holdable_locations? Similar to `process_volumetric_calls`?

Reply:
It seems like `@holdable_locations` is used only for `filter_holdables` (if I am not missing sth), maybe then getting rid of line 15 here and just doing below would work?

```
 def filter_holdables
      holdable_locations = find_holdable_locations
      @call_list.filter! do |c|
        current_locations = c.items.map(&:current_location)
        current_locations.filter { |loc| holdable_locations.include? loc }.present?
      end
    end
```

Reply:
I think that's a cleaner code version, but, it would likely be less efficient because that `holdable_locations` method would get called many times rather than just once, right?

Reply:
`filter_holdables` called by `process_volumetric_calls` which is called by `generate`, so I thought it would not be many times but maybe I missing sth?

Reply:
Okay, yep I see. I do like this better I think. I'll make the change. If there's one thing this gives up it's that the need to conduct this extra external class call (i.e., dependency) is a bit buried more. But, I think it's okay and this is more efficient.